src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -329,10 +329,19 @@ public async Task<bool> DeleteSecretAsync(string secretName, string keyScope, Sc
                 return secrets;
             }, secretsFactory).ContinueWith(t =>
             {
-                if (t.IsFaulted && t.Exception.InnerException is RequestFailedException)
+                if (t.IsFaulted)
                 {
-                    result = OperationResult.Forbidden;
+                    _logger.LogError("Error adding or updating secrets", t.Exception.InnerException);
+
+                    result = OperationResult.Error;
+
+                    if (t.Exception.InnerException is RequestFailedException)
```

Comment:
I don't think we need to retain this special 403 conversion code. @liliankasem updated this logic in [this commit](https://github.com/Azure/azure-functions-host/commit/a642d6d3aadeb05eff1e2031fc8c7824b32b97fb) - it used to be scoped specifically to a KeyVaultErrorException, but was changed to the more general RequestFailedException.

Reply:
agreed, @liliankasem any concerns changing error response to 500?

Reply:
Ohhh I remember this. With the change to `Azure.Identity`, they return a `RequestFailedException` instead of a `KeyVaultErrorException` so that had to change. I still want us to return 403 if the exception was due to being forbidden, for anything else you can return 500.

Reply:
I wonder if we need to check the status code somewhere from the initial request to make that happen

Reply:
Based on discussion of the original code in PR https://github.com/Azure/azure-functions-host/pull/8361, I think we're agreeing that we shouldn't have been doing the conversion in the first place, so the special handling will go away in both cases.

Reply:
Sounds good, then let's remove it!

Reply:
@liliankasem I had some further discussion with mathew, I realizewd that returning 403 would be incorrect because to the user that would mean the call to generate the key does not have the correct auth, which is not true. If we fail to generate / write a key we should be returning 500. Have updated my code to reflect the same.